go-controller/pkg/factory/factory.go

```diff
@@ -347,8 +349,10 @@ func NewNodeWatchFactory(ovnClientset *util.OVNClientset, nodeName string) (*Wat
 }
 
 func (wf *WatchFactory) WaitForWatchFactoryStopChannel(stopChan chan struct{}) {
+	wf.wg.Add(1)
 	<-wf.stopChan
 	close(stopChan)
+	wf.wg.Done()
```

Comment:
@bpickard22 I was thinking a little more after we talked on this.
I think we need to move the `wf.wg.Add(1)` + `wf.wg.Done()` to `func (r *RetryFramework) periodicallyRetryResources() {`, so that the waitgroup blocks until that function returns. Do you agree? Can you make that change?

Reply:
So the `wg.Add(1)` should be called by the function that launches the goroutine right before it launches it, while `wg.Done()` should be called when the goroutine exits. Here's a good example for reference that we can apply here too: https://gobyexample.com/waitgroups

In this case we should make the waitgroup in WatchFactory public. 

Finally, what I was saying in one of my previous comments is that, with these changes, the `StopChannel` attributes in `RetryFramework` is no longer used and should be removed: https://github.com/ovn-org/ovn-kubernetes/blob/c72e852628872cd75d1fb5db26fc758ee7d91bdc/go-controller/pkg/retry/obj_retry.go#L79